### PR TITLE
make `value` property can be null

### PR DESCRIPTION
**Pull request by springuper**
change `this.props.value == undefined` to `typeof this.props.value == 'undefined'` because I have a use case where the value can be null.

My case is, to implement `RIEJson` component which extends `RIEStatefulBase`, but the `value` can be `null` initially. So I write like this:

```js
class RIEJson extends RIEStatefulBase {
  static propTypes = {
    value: React.PropTypes.oneTypeOf([
      React.PropTypes.array,
      React.PropTypes.object,
    ]),
  };

  ...
}
```

### CODE REVIEW

**Comment by springuper** [comment]
I don't know why there are so many built file changes. I'm using Node v6.9.1

File: lib/RIEBase.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ var RIEBase = function (_React$Component) {
     function RIEBase(props) {
         _classCallCheck(this, RIEBase);
 
-        var _this = _possibleConstructorReturn(this, Object.getPrototypeOf(RIEBase).call(this, props));
+        var _this = _possibleConstructorReturn(this, (RIEBase.__proto__ || Object.getPrototypeOf(RIEBase)).call(this, props));
```
Comment (springuper) [inline lib/RIEBase.js]: seems that it's caused by different babel version.

**Comment by devvmh** [comment]
I'm currently trying to make markdown-enabled editing, so I'm hoping to use dangerouslySetInnerHTML, and this PR might help me

**Comment by kaivi** [comment]
I'm changing the line manually to avoid conflict resolution on `demo.js` artifact.

**Pull request closed without merging**